invenio/base/scripts/apache.py

```diff
@@ -17,194 +17,414 @@
+                      file=sys.stderr)
+                wsgi['user'] = current_user
+
+            try:
+                wsgi['dir'] = app.config['CFG_WSGIDIR']
+            except KeyError:
+                wsgi_dir = path.abspath(
+                    pkg_resources.resource_filename('invenio', '')
+                )
+                print('>>> Variable {} not set. Using {}'
+                      .format('CFG_WSGIDIR', wsgi_dir),
+                      file=sys.stderr)
+                wsgi['dir'] = wsgi_dir
+
+            wsgi['script_alias'] = '/wsgi'
+            # TODO: Test if this is useful anymore
+            wsgi['socket_directive_needed'] = {
+                self._candidate_paths['apache']: False,
+                self._candidate_paths['apache2']: False,
+                self._candidate_paths['tls']: True
+                }[self._resolved_dir]
+            wsgi['process_name'] = app.config['CFG_APACHE_WSGI_PROCESS_NAME']
+            wsgi['process_group'] = app.config['CFG_APACHE_WSGI_PROCESS_GROUP']
+            wsgi['socket_prefix'] = path.join(self.prefix, 'var', 'run')
+            wsgi['script_fullpath'] = path.join(wsgi['dir'], 'invenio.wsgi')
+            return wsgi
+
+    apache_cfg = ApacheConfig()
+
+    # Resolve requested schemes
+    if enable_ssl:
+        schemes = ('http', 'https')
+    else:
+        schemes = ('http',)
+    if apache_cfg.vhosts['http'].port == apache_cfg.vhosts['https'].port:
+        schemes = ('http',)
+
+    conf_file = 'apache-vhost.conf'
+    # Create conf dir
+    apache_conf_dir = path.join(app.instance_path, 'apache')
+    dir_util.mkpath(apache_conf_dir)
+    # Backup old file
+    apache_vhost_file = path.join(apache_conf_dir, conf_file)
+    if path.exists(apache_vhost_file):
+        shutil.copy(apache_vhost_file,
+                    apache_vhost_file + '.bak')
+    # Write new file
+    with open(apache_vhost_file, 'w') as f:
+        out = render_template_to_string("{}.tpl".format(conf_file),
+                                        cfg=apache_cfg,
+                                        schemes=schemes)
+        print(out, file=f)
 
     print(wrap_text_in_a_box("""\
-Apache virtual host configuration file(s) for your Invenio site
-was(were) created.  Please check created file(s) and activate virtual
-host(s).  For example, you can put the following include statements in
-your httpd.conf:\n
+Apache virtual host configuration file for your site was created.
+You may enable them by adding the following to your httpd.conf:\n
 
-%s
+Include {}
 
 Please see the INSTALL file for more details.
-    """ % '\n\n'.join(tuple(map(
-        lambda x: "Include " + apache_conf_dir.encode('utf-8') + os.sep + x,
-        list(conf_files[:1 if no_ssl else 2]))))))
-    print(">>> Apache conf files created.")
+                             """.format(apache_vhost_file)))
 
 
 def main():
     from invenio.base.factory import create_app
-    app = create_app()
```

Comment:
Do you have any reason for changing these 2 lines? If there is serious reason we should consider amending other manage scripts too.